pkg/network/ovn_kubernetes_test.go

```diff
@@ -139,6 +141,66 @@ func TestRenderOVNKubernetesIPv6(t *testing.T) {
 	g.Expect(script).To(ContainSubstring("pssl:9641:[::]"))
 }
 
+func TestRenderedOVNKubernetesConfig(t *testing.T) {
+	type testcase struct {
+		desc                string
+		expected            string
+		hybridOverlayConfig *operv1.HybridOverlayConfig
+	}
+	testcases := []testcase{
+		{
+			desc: "default",
+			expected: "[default]\nmtu=\"1500\"\ncluster-subnets=\"10.128.0.0/15/23,10.0.0.0/14/24\"\n" +
+				"encap-port=\"8061\"\n\n[kubernetes]\nservice-cidrs=\"172.30.0.0/16\"\n" +
+				"ovn-config-namespace=\"openshift-ovn-kubernetes\"\napiserver=\"" + fmt.Sprintf("https://%s:%s", os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")) + "\"\n\n" +
+				"[gateway]\nmode=local\nnodeport=true",
+		},
+		{
+			desc: "HybridOverlay",
+			expected: "[default]\nmtu=\"1500\"\ncluster-subnets=\"10.128.0.0/15/23,10.0.0.0/14/24\"\n" +
+				"encap-port=\"8061\"\n\n[kubernetes]\nservice-cidrs=\"172.30.0.0/16\"\n" +
+				"ovn-config-namespace=\"openshift-ovn-kubernetes\"\napiserver=\"" + fmt.Sprintf("https://%s:%s", os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")) + "\"\n\n" +
+				"[gateway]\nmode=local\nnodeport=true\n\n[hybridoverlay]\nenabled=true\ncluster-subnets=\"10.132.0.0/14\"",
+			hybridOverlayConfig: &operv1.HybridOverlayConfig{
+				HybridClusterNetwork: []operv1.ClusterNetworkEntry{
+					{CIDR: "10.132.0.0/14", HostPrefix: 23},
+				},
+			},
+		},
```

Comment:
and two more cases to test the vxlan and enabled-but-no-clusternetworks cases?

Reply:
done